File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def convertBinary(num):
    result = ""
    while num:
        n, m = divmod(num, 2)
        result += str(m)
        num = n
    return result[::-1]

def solution(n):
    answer = 0
    nToBinary = convertBinary(n)
    nOneCount = nToBinary.count("1")
    target = n + 1
    
    while True:
        targetToBinary = convertBinary(target)
        targetOneCount = targetToBinary.count("1")
        if nOneCount == targetOneCount:
            return target
        target += 1
    

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    let target = n;
    while (true) {
        // 조건 1;
        target++;
        
        // 조건 2
        const nOneCount = n.toString(2).split("").filter((i) => i === "1").length;
        const targetOneCount = target.toString(2).split("").filter((i) => i === "1").length;
        
        if (nOneCount === targetOneCount) return target;
    }
}